scripts/survived.py

```
@@ -0,0 +1,314 @@
+                pool.setdefault(s, []).append(str(i))
+
+    kept, absent = 0, []
+    for l in lines:
+        hit = next((ln for ln in pool.get(l, []) if (path, ln) not in claimed), None)
+        if hit is not None:
+            claimed.add((path, hit))
+            kept += 1
+        else:
+            absent.append(l)
+
+    moved = sum(1 for l in absent if find_elsewhere(ref, l, path, claimed))
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)